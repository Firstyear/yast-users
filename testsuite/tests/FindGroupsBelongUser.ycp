{

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[
	"target" : $[
	    "bash_output" : $[],
	],
    ];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);

    import "Users";

    Users::groups ["local" ] = $[
        100: $[ "groupname": "users", "userlist":"", "more_users": "hh","type": "local" ],
        17:  $[ "groupname": "audio", "userlist":"hh,ab", "more_users": "","type": "local" ]
    ];

    Users::groups ["ldap" ] = $[
        100: $[ "groupname": "users", "uniqueMember": [ "uid=hh" ],
		"more_users": "u1,u2","type": "ldap" ]
    ];

    DUMP (sformat ("local groups::\n %1", Users::groups ["local"]:$[]));

    DUMP (sformat ("ldap groups::\n %1", Users::groups ["ldap"]:$[]));

    DUMP ("==========================================================");

    string user = "hh";
    string dn = "uid=hh";
    DUMP ( sformat("user %1 is in groups:", user) );

    TEST(``(Users::FindGroupsBelongUser (user, dn)), [], 0);

    DUMP ("==========================================================");

    user = "ab";
    DUMP ( sformat("user %1 is in groups:", user) );

    TEST(``(Users::FindGroupsBelongUser (user, "")), [], 0);

    DUMP ("==========================================================");

    user = "abc";
    DUMP ( sformat("user %1 is in groups:", user) );

    TEST(``(Users::FindGroupsBelongUser (user, "")), [], 0);

}
