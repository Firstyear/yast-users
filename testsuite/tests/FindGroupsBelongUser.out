Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Execute	.target.bash_output "locale -k charmap" $[]
Dump	local groups::
Dump	 $[17:$["groupname":"audio", "more_users":"", "type":"local", "userlist":"hh,ab"], 100:$["groupname":"users", "more_users":"hh", "type":"local", "userlist":""]]
Dump	ldap groups::
Dump	 $[100:$["groupname":"users", "more_users":"u1,u2", "type":"ldap", "uniqueMember":["uid=hh"]]]
Dump	==========================================================
Dump	user hh is in groups:
Return	users,audio
Dump	==========================================================
Dump	user ab is in groups:
Return	audio
Dump	==========================================================
Dump	user abc is in groups:
Return	
